pkg/endpoint/bpf.go

```diff
@@ -648,12 +656,15 @@ func (e *Endpoint) regenerateBPF(owner Owner, epdir, reason string) (revnum uint
 	rundir := owner.GetStateDir()
 	debug := strconv.FormatBool(viper.GetBool(option.BPFCompileDebugName))
 
+	stats.prepareBuild.End()
+
 	if bpfHeaderfilesChanged {
```

Comment:
The removal of `BuildMutex` means we no longer hold any lock while we are compiling the program for an endpoint. Is this safe? Are we relying upon the logic to ensure that only one regeneration per endpoint is taking place at a given time? : 

https://github.com/cilium/cilium/blob/d8314c7734c6efee94411bf6e506b91ac853d0e7/pkg/endpoint/policy.go#L772-L828

Reply:
I checked this again. While we ensure that only one endpoint can be queued for regeneration. When it dequeued for building, parallel builds could indeed happen. I'll move the removal of the BuildMutex to a separate PR to make review easier.